nexus/src/app/sagas/disk_create.rs

```diff
@@ -155,16 +155,36 @@ async fn sdc_create_disk_record(
                         .await
                         .map_err(ActionError::action_failed)?;
 
-            global_image.block_size
-        }
-        params::DiskSource::ImportingBlocks { block_size } => {
-            db::model::BlockSize::try_from(*block_size).map_err(|e| {
-                ActionError::action_failed(Error::internal_error(
-                    &e.to_string(),
-                ))
-            })?
-        }
-    };
+                db_snapshot.block_size
+            }
+            params::DiskSource::Image { image_id } => {
+                let (.., image) =
+                    LookupPath::new(&opctx, &osagactx.datastore())
+                        .image_id(*image_id)
+                        .fetch()
+                        .await
+                        .map_err(ActionError::action_failed)?;
```

Comment:
The `ImportingBlocks` path does a `map_error(|e| .. e.to_string() ...`. Why not do that here and for GlobalImage path too?